device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/IotHubTransportMessage.java

```diff
@@ -293,6 +293,11 @@ public String toString()
             base.append("Request Id [").append(this.requestId).append("] ");
         }
 
+        if (this.getDeviceOperationType() != null && this.getDeviceOperationType() != DeviceOperations.DEVICE_OPERATION_UNKNOWN)
+        {
+            base.append("Device Operation Type [").append(this.getDeviceOperationType()).append("] ");
```

Comment:
This information has been useful to me when debugging these flakey tests, and I suspect customer's will appreciate this too

Reply:
This will log if a message is a twin subscription message, method subscription message, etc.